packages/server/ts/game/entity/character/player/warp.ts

```diff
@@ -24,31 +28,32 @@ class Warp {
             return;
         }
 
-        let data = Modules.Warps[id];
+        let data = this.map.getWarpById(id);
 
         if (!data)
             return;
 
-        let name = data[0],
-            x = data[3] ? data[1] + Utils.randomInt(0, 1) : data[1],
-            y = data[3] ? data[2] + Utils.randomInt(0, 1) : data[2],
-            levelRequirement: number = data[4];
-
         if (!this.player.finishedTutorial()) {
-            this.player.notify('You cannot warp while in this event.');
+            this.player.notify('You cannot warp while in the tutorial.');
             return;
         }
 
-        if (!this.hasRequirement(levelRequirement)) {
-            this.player.notify('You must be at least level ' + levelRequirement + ' to warp here!');
+        if (!this.hasRequirement(data.level)) {
+            this.player.notify(`You must be at least level ${data.level} to warp here!`);
             return;
         }
 
-        this.player.teleport(x, y, false, true);
+        // data.x += Utils.randomInt(-1, 1);
+        // data.y += Utils.randomInt(-1, 1);
```

Comment:
Let's not leave a commented out code. If the commented code is an idea to which we want to come back later then we need to add more description to it. Otherwise, I would remove it.